### PR TITLE
[function](nereids) support array_apply/array_repeat/group_uniq_array/ipv4numtostring

### PR DESCRIPTION
**Pull request by englefly**
## Proposed changes
nereids  support functions: array_apply/array_repeat/group_uniq_array/ipv4numtostring
Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by englefly** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/ArrayApply.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+
+/**
+ * scalar function array_apply
+ */
+public class ArrayApply extends ScalarFunction
+        implements BinaryExpression, ExplicitlyCastableSignature, PropagateNullable {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/ArrayApply.java:40]: the second arg could be slot or only accept literal? do we need to check for legality

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/ArrayApply.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                            new FollowToAnyDataType(0)));
+
+    public ArrayApply(Expression arg0, Expression arg1, Expression arg2) {
+        super("array_apply", arg0, arg1, arg2);
+    }
+
+    @Override
+    public ArrayApply withChildren(List<Expression> children) {
+        Preconditions.checkArgument(children.size() == 3,
+                "array_apply accept 3 args, but got %d (%s)",
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/ArrayApply.java:53]: if I remember correctly, check template arguments only support %s

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/ArrayRepeat.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import java.util.List;
+
+/**
+ * ScalarFunction ArrayRepeat
+ */
+public class ArrayRepeat extends ScalarFunction
+        implements BinaryExpression, ExplicitlyCastableSignature, AlwaysNotNullable {
+
+    public static final List<FunctionSignature> SIGNATURES = ImmutableList.of(
+            FunctionSignature.ret(ArrayType.of(new AnyDataType(0))).args(new AnyDataType(0), BigIntType.INSTANCE)
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/ArrayRepeat.java:42]:
```suggestion
            FunctionSignature.ret(ArrayType.of(new FollowToAnyDataType(0))).args(new AnyDataType(0), BigIntType.INSTANCE)
```

File: gensrc/script/doris_builtins_functions.py [hunk 4]
```diff
@@ -493,15 +493,15 @@
         [['array_apply'], 'ARRAY_SMALLINT', ['ARRAY_SMALLINT', 'VARCHAR', 'SMALLINT'], ''],
         [['array_apply'], 'ARRAY_INT',      ['ARRAY_INT', 'VARCHAR', 'INT'], ''],
         [['array_apply'], 'ARRAY_BIGINT',   ['ARRAY_BIGINT', 'VARCHAR', 'BIGINT'],   ''],
-        [['array_apply'], 'ARRAY_LARGEINT', ['ARRAY_LARGEINT', 'VARCHAR', 'ARRAY_LARGEINT'], ''],
+        [['array_apply'], 'ARRAY_LARGEINT', ['ARRAY_LARGEINT', 'VARCHAR', 'LARGEINT'], ''],
```
Comment (morrySnow) [inline gensrc/script/doris_builtins_functions.py:496]: please push a new PR only include legacy planner fix into branch-2.0

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/ArrayRepeat.java (hunk 3 above)
Comment (XieJiann) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/ArrayRepeat.java:42]: the 2nd AnyDataType should be AnyDataType.INSTANCE_WITHOUT_INDEX

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/ArrayApply.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+import java.util.List;
+
+/**
+ * scalar function array_apply
+ */
+public class ArrayApply extends ScalarFunction
+        implements BinaryExpression, ExplicitlyCastableSignature, PropagateNullable {
+    public static final List<FunctionSignature> SIGNATURES = ImmutableList.of(
+            FunctionSignature.retArgType(0)
+                    .args(ArrayType.of(new AnyDataType(0)), VarcharType.SYSTEM_DEFAULT,
```
Comment (XieJiann) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/ArrayApply.java:43]: use AnyDataType.INSTANCE_WITHOUT_INDEX

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/ArrayApply.java [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+                            new FollowToAnyDataType(0)));
```
Comment (XieJiann) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/ArrayApply.java:44]: ArrayType.of(AnyDataType.WITH_OUT_INDEX)

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/ArrayApply.java (hunk 1 above)
Comment (englefly) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/ArrayApply.java:40]: array_apply(arr, op, val)
op and val only support literal. add check in constructor

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/ArrayApply.java (hunk 2 above)
Comment (englefly) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/ArrayApply.java:53]: yes, you are correct

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/ArrayApply.java (hunk 6 above)
Comment (englefly) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/ArrayApply.java:44]: the index is used by the last arg

**Review by keanji-x — APPROVED** [review]

File: gensrc/script/doris_builtins_functions.py (hunk 4 above)
Comment (englefly) [reply gensrc/script/doris_builtins_functions.py:496]: https://github.com/apache/doris/pull/25329

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Pull request merged**